### PR TITLE
add anchor as a new component to support anchor-button scenario

### PR DESCRIPTION
**Pull request by chrisdholt**
## Naming
For a variety of reasons, supporting two separate semantic elements within the ShadowDOM of a web component is less than ideal. With that, we need to find an approach to support scenarios for elements which functionally and semantically are anchors, but appear like a button.

From a verbosity standpoint, "anchor button" is the semantic that we tend to use to describe these. Fluent UI React v9 utilizes "as" to shorthand this assignment. Given this needs to be a separate HTML element, I'm proposing a shorter name than `anchor-button` and utilizing a class and tag name of simply `<fluent-anchor>`. A few key reasons for this:
- Anchor already exists in Fluent UI Web Components (stable) using the same semantic, construct, and approach. 
- Anchor is shorter to write and less verbose considering devx and convenience
- There is no conflicting namespace. "Anchor" isn't a construct in Fluent components and "link" conveys a different semantic meaning. 

Overall, it seems reasonable to simply go with the shorter name as it maintains an existing approach, there are no semantic conflicts in Fluent, and it's just plain easier and shorter to write.

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d901fa95e8f90c1b5dabda8602a72ecb77288f0c'>d901fa95e8f90c1b5dabda8602a72ecb77288f0c</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/web-components/src/anchor/anchor.styles.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { css } from '@microsoft/fast-element';
+import { styles as ButtonStyles } from '../button/button.styles.js';
+
+// Need to support icon hover styles
+export const styles = css`
+  ${ButtonStyles}
+
+  .content {
+    text-align: center;
```
Comment (eljefe223) [inline packages/web-components/src/anchor/anchor.styles.ts:9]: curious why this is different from button?

Comment (chrisdholt) [reply packages/web-components/src/anchor/anchor.styles.ts:9]: We can add it to button if we want, but it doesn't make a bit of difference to the actual button code. I'm open here. I kept it separate to not make a pattern of hoisting deltas into a single stylesheet, especially if we're going to reuse. With the perf gate up we can look to identify if this is as performant or less than extracting the entire button styles to a css partial.

Comment (chrisdholt) [reply packages/web-components/src/anchor/anchor.styles.ts:9]: Update here - given that compound button text *is not* centered, I don't think we want to add this to button as of yet.

**Review by eljefe223 — APPROVED** [review]

File: packages/web-components/src/anchor/anchor.stories.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+const storyTemplate = html<AnchorStoryArgs>`
+  <fluent-anchor
+    href="${x => x.href}"
+    appearance="${x => x.appearance}"
+    shape="${x => x.shape}"
+    size="${x => x.size}"
+    ?disabled="${x => x.disabled}"
+    ?disabled-focusable="${x => x.disabledFocusable}"
+    ?icon-only="${x => x.iconOnly}"
+    ?icon="${x => x.icon}"
```
Comment (miroslavstastny) [inline packages/web-components/src/anchor/anchor.stories.ts:20]: After the latest changes in `fluent-button`, there is no `icon` attribute, right? Remove it everywhere.

**Review by miroslavstastny — COMMENTED** [review]
> There is no conflicting namespace. "Anchor" isn't a construct in Fluent components and "link" conveys a different semantic meaning.

So we will have `fluent-anchor` (which looks like a button) and intrinsic `a`. If we ever add `fluent-link` my head will explode.

**Comment by chrisdholt** [comment]
> > There is no conflicting namespace. "Anchor" isn't a construct in Fluent components and "link" conveys a different semantic meaning.
> 
> So we will have `fluent-anchor` (which looks like a button) and intrinsic `a`. If we ever add `fluent-link` my head will explode.

I guess “link” is a legit instance and appearance we need to adopt isn’t it? I’m open to “anchor-button” but also, yuck!

**Comment by miroslavstastny** [comment]
> > > There is no conflicting namespace. "Anchor" isn't a construct in Fluent components and "link" conveys a different semantic meaning.
> > 
> > 
> > So we will have `fluent-anchor` (which looks like a button) and intrinsic `a`. If we ever add `fluent-link` my head will explode.
> 
> I guess “link” is a legit instance and appearance we need to adopt isn’t it? I’m open to “anchor-button” but also, yuck!

I would understand the `anchor-button` more (because that's what it is). But will leave it up to you.

**Review by miroslavstastny — APPROVED** [review]

**Pull request merged**